src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -119,6 +122,8 @@ private void HandleOrRecordException(Exception e)
             }
 
             ConnectionString = runtimeConfig.GetDataSourceFromDataSourceName(dataSourceName).ConnectionString;
+            SqlConnectionStringBuilder connectionStringBuilder = new(ConnectionString);
+            DatabaseName = connectionStringBuilder.InitialCatalog;
```

Comment:
For MsSQL, it looks like `InitialCatalog` would always get populated. For other SQL database types, does it get populated always (it can be the case, but need to go over the docs and confirm that)

_A slightly different approach to consider for fetching database name:_

`DatabaseName` property is used inside the function `GetTableWithSchemaFromDataSetAsync`. In that function, there is a call to a method `FillSchemaForTableAsync`. Inside there, a connection to the database is opened, post that, the  
database name is fetched from the connection object using `conn.Database` L:1364. When database name is extracted this way (after successfully connecting to the actual database), we can be sure that a) Connection string is valid b) Database specific behavior (if any) w.r.t to how connection string is interpreted is not going to have any effect.

Reply:
I initially wanted to use this approach, but it means opening up a database connection only to parse out the database name, which is not very efficient. Correctly and carefully parsing the connection string to pull out the database name should be a lot faster and also entirely synchronous.

Reply:
Since, there is going to be a separate SqlMetadataProvider class for every database, it should be fine to not include the database name. Distinguishing the backing objects based on schema name + table/view/stored-procedure name should be sufficient. 

Since, this entire thread is focused on diferent approaches for fetching the database names, this is not applicable anymore. So, resolving this thread.